providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/connector/file/FileConnector.java

```diff
@@ -9,36 +9,73 @@
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
+import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.concurrent.BlockingQueue;
 import java.util.concurrent.LinkedBlockingQueue;
 
 /**
- * File connector reads flag configurations and expose the context through {@code Connector} contract.
+ * File connector reads flag configurations from a given file, polls for changes and expose the content through
+ * {@code Connector} contract.
  * The implementation is kept minimal and suites testing, local development needs.
  */
 @SuppressFBWarnings(value = {"EI_EXPOSE_REP", "PATH_TRAVERSAL_IN"},
         justification = "File connector read feature flag from a file source.")
 @Slf4j
 public class FileConnector implements Connector {
 
+    private static final int POLL_INTERVAL_MS = 5000;
+
     private final String flagSourcePath;
     private final BlockingQueue<StreamPayload> queue = new LinkedBlockingQueue<>(1);
+    private boolean shutdown = false;
 
     public FileConnector(final String flagSourcePath) {
         this.flagSourcePath = flagSourcePath;
     }
```

Comment:
we could also add another constructor to customize the polling interval
```
public FileConnector(final String flagSourcePath, final int poll_interval_ms) 
```

Reply:
This is useful only if we plan to expose the polling interval as a configuration option. However, I think this is not a priority right now hence I think we can stick with hardcoded value (5000ms) till we see the need.